java/memory/src/main/java/org/apache/arrow/memory/BaseAllocator.java

```diff
@@ -143,6 +143,20 @@ public static int nextPowerOfTwo(int val) {
     }
   }
 
+  /**
+   * Rounds down the provided value to the nearest power of two.
+   *
+   * @param val An integer value.
+   * @return The closest power of two of that value.
+   */
+  public static int prevPowerOfTwo(int val) {
```

Comment:
please add tests explicitly for this method?

Reply:
@emkornfield Good suggestion. I have added tests for this method and nextPowerOfTwo.